Fix 1058 runqing

runqingz opened the pull request:
**IMPORTANT**: By submitting a patch, you agree that your work will be licensed under the [license](https://github.com/umple/umple/blob/master/LICENSE.md) used by the project.

## Description

This is the PR for issue #1058. This PR fix the issue that the error message of error number 19 is not informative enough. This fix will also make the message suggesting potential fix to the ump file to user. 

## Tests

A few tests is also added to make sure the error raise in the correct condition.

runqingz committed:
Fix on the Error message

runqingz committed:
Merge branch 'master' into fix_1058_runqing

runqingz committed:
Fix the error message so it can correctly show class name

runqingz committed:
Merge remote-tracking branch 'umple/master' into fix_1058_runqing

runqingz committed:
Merge remote-tracking branch 'umple/master' into fix_1058_runqing

vahdat-ab commented on cruise.umple/src/en.error:
```diff
@@ -22,7 +22,7 @@
 17: 2, "http://cruise.eecs.uottawa.ca/umple/E017NonImmutableBidirectionality.html", Two-way associations may not be immutable ;
 18: 2, "http://cruise.eecs.uottawa.ca/umple/E018ReflexiveImmutable.html", Reflexive immutable associations may not be mandatory ;
-19: 2, "http://cruise.eecs.uottawa.ca/umple/E019DuplicateAssociation.html", Class '{0}' may not have multiple associations with the same name '{1}' ;
+19: 2, "http://cruise.eecs.uottawa.ca/umple/E019DuplicateAssociation.html", There are multiple associations between class '{0}' and class '{1}'. Role names need to be added at '{0}' side to distinguish the different association ends in that class ;
```
add a full stop to the end of the sentence.

vahdat-ab commented on cruise.umple/test/cruise/umple/compiler/024_multiAssocToSameClassWithOneRoleName.ump:
```diff
@@ -0,0 +1,7 @@
+class Course{
+  * courseProf -- 1 Person prof;
+  * -- 1..* Person students;
+}
```
consider testing more example in the both sides.
For example, 
```
class A{
  0..1  a1-- * B;
  0..1  a2-- * B;
}
class B{}
```
will give the following message in which we have a asociation between `A` and  `A`.
```
Error 19 on line 3 of file "test.ump":
There are multiple associations between class 'A' and class 'A'. Role names need to be added at 'A' side to distinguish the different association ends in that class

```

vahdat-ab replied:
Any improvement on this? The association should be between `A` and `B`.

vahdat-ab commented on cruise.umple/test/cruise/umple/compiler/024_multiAssocToSameClassWithMultiRoleName.ump:
```diff
@@ -0,0 +1,7 @@
+class Course{
+  * courseProf -- 1 Person prof;
+  * courseStudent -- 1..* Person students;
+}
```
Your code doesn't pas the following example:
```
associationClass Friendship {
  * inviterFriend Profile invitee;
  * inviteeFriend Profile inviter;
  status;
}
class Profile {
    name;
}
```
maybe you wanted to have another PR for it?

runqingz replied:
Yes, that will be another PR

runqingz committed:
Added more test fixture

runqingz committed:
Fix the issue in PR comment

class A{
  0..1  a1-- * B;
  0..1  a2-- * B;
}
class B{}
should give message similar to
Error 19 on line 3 of file "test.ump":
There are multiple associations between class 'A' and class ‘B’. Role
names need to be added at ‘B’ side to distinguish the different
association ends in that class

runqingz committed:
Fixed further issue with error. Added further tests to test the actual error message

runqingz committed:
Fix the risk that check duplicate associations can generate multiple error message

runqingz committed:
Merge remote-tracking branch 'umple/master' into fix_1058_runqing
